Testing/SystemTests/tests/framework/BASISTest.py

```
@@ -314,6 +316,69 @@ def validate(self):
+        mask = FileFinder.getFullPath("arg_si.dat")
+        assert os.path.exists(mask)
+        with self.assertRaisesRegex(RuntimeError, f"Mask file {mask} is not of the correct format"):
+            BASISReduction(**def_args, MaskFile=mask)
+
+        # lookup the module object for BASISReduction to patch the lookups to simulate situations where a mask is not found
+        mod = sys.modules.get("BASISReduction", None)
+        if mod is None:
+            raise RuntimeError("Could not locate BASISReduction module to patch default mask lookups")
+
+        # step 3: run with no mask -- should resort to default, which will not exist
+        with tempfile.TemporaryDirectory() as dir:
```

Ruff (0.15.10)

[error] Variable `dir` is shadowing a Python builtin

(A001)